app/models/blog_post.rb

```diff
@@ -2,8 +2,13 @@
 class BlogPost < ApplicationRecord
   belongs_to :blog
   belongs_to :author, class_name: 'User',
-                      foreign_key: 'user_id',
-                      inverse_of: 'blog_posts'
+                      foreign_key: :user_id,
+                      inverse_of: :blog_posts
+
+  has_one :discussion, as: :resource, dependent: :destroy
+
+  delegate :discussion_hidden, to: :discussion, allow_nil: true
+  delegate :discussion_locked, to: :discussion, allow_nil: true
```

Comment:
Consider using:

```
delegate :hidden, :locked, to: :discussion, prefix: true, allow_nil: true
```

The `prefix` option means the generated methods will still be named `discussion_hidden` and `discussion_locked`, and those methods in `Discussion` can be deleted.

If you prefer the `predicate?` method naming:

```
delegate :hidden?, :locked?, to: :discussion, prefix: true, allow_nil: true
```

would generate `BlogPost#discussion_hidden?` and `#discussion_locked?` methods.